Makefile

```diff
@@ -9,7 +9,7 @@ bins: temporal-server temporal-cassandra-tool temporal-sql-tool tdbg
 all: update-tools clean proto bins check test
 
 # Used by Buildkite.
-ci-build: bins build-tests ci-update-tools shell-check copyright-check proto go-generate gomodtidy ensure-no-changes
+ci-build-misc: bins ci-update-tools shell-check copyright-check proto go-generate gomodtidy ensure-no-changes
```

Comment:
why this change?

Reply:
The "build-tests" part was basically just a quick check for compilation errors to avoid running the rest of the test suite (and then it threw away the results of course). Now there's a initial "build" step that builds the functional tests and all the parts of the server that they depend on (most of it), which basically serves the same purpose. The reason to build only functional tests and not unit tests in the build step is that the functional test compilation results will be reused by many steps, and steps that are on the critical path, while the unit test compilation results are only used by a single step, and that one is not on the critical path so having it do more work doesn't delay the overall run.